sriovnet.go

```diff
@@ -453,3 +453,24 @@ func GetNetDevicesFromPci(pciAddress string) ([]string, error) {
 	pciDir := filepath.Join(PciSysDir, pciAddress, "net")
 	return getFileNamesFromPath(pciDir)
 }
+
+// Transform a netdevice name into its PCI address, if applicable.
+func GetPciFromNetDevice(name string) (string, error) {
+	devPath := filepath.Join(NetSysDir, name)
+
+	realPath, err := utilfs.Fs.Readlink(devPath)
+	if err != nil {
+		return "", fmt.Errorf("device %s not found: %s", name, err)
+	}
+
+	parent := filepath.Dir(realPath)
+	base := filepath.Base(parent)
+	for parent != "/" && !pciAddressRe.MatchString(base) {
```

Comment:
what is this trying to address ? SF ?

can you add some comment explaining when we would enter this if statement ?

Reply:
Actually, more than just SF, but indeed it adds the support for SF that way.

There is no 'EvalSymlinks' in afero, so no way to properly mock up 'readlink -f' -- once we access `/sys/class/net/<name>`, we cannot actually resolve  the link. We can read the value of its target with 'Readlink', but not evaluate it into the actual node in the sysfs.

To do the checks I had previously, verifying that the interface has a PCI device somewhere, then instead we must go up the chain and try to match each time.

Additionally, it just so happens that SF are 3 levels above this link target, while other devices will be 2 levels above. Allowing to parse up to `/` adds support for SFs.

I'll add the relevant comments.

Reply:
i see, could you also add example of sysfs paths for both VF, PF and SF associated netdevs ? for clarity